cloudinit/sources/DataSourceAzure.py

```diff
@@ -1338,76 +1338,101 @@ def load_azure_ds_dir(source_dir):
-                    if addr_type == 'ipv4':
-                        default_prefix = '24'
-                    else:
-                        default_prefix = '128'
-                        if addresses:
-                            dev_config['dhcp6'] = True
-                            # non-primary interfaces should have a higher
-                            # route-metric (cost) so default routes prefer
-                            # primary nic due to lower route-metric value
-                            dev_config['dhcp6-overrides'] = dhcp_override
-                    for addr in addresses[1:]:
-                        # Append static address config for ip > 1
-                        netPrefix = intf[addr_type]['subnet'][0].get(
-                            'prefix', default_prefix)
-                        privateIp = addr['privateIpAddress']
-                        if not dev_config.get('addresses'):
-                            dev_config['addresses'] = []
-                        dev_config['addresses'].append(
-                            '{ip}/{prefix}'.format(
-                                ip=privateIp, prefix=netPrefix))
-                if dev_config:
-                    mac = ':'.join(re.findall(r'..', intf['macAddress']))
-                    dev_config.update({
-                        'match': {'macaddress': mac.lower()},
-                        'set-name': nicname
-                    })
-                    # With netvsc, we can get two interfaces that
-                    # share the same MAC, so we need to make sure
-                    # our match condition also contains the driver
-                    driver = device_driver(nicname)
-                    if driver and driver == 'hv_netvsc':
-                        dev_config['match']['driver'] = driver
-                    netconfig['ethernets'][nicname] = dev_config
-            evt.description = "network config from imds"
-        else:
-            blacklist = ['mlx4_core']
-            LOG.debug('Azure: generating fallback configuration')
-            # generate a network config, blacklist picking mlx4_core devs
-            netconfig = net.generate_fallback_config(
-                blacklist_drivers=blacklist, config_driver=True)
-            evt.description = "network config from fallback"
-        return netconfig
+    if imds_metadata != sources.UNSET and imds_metadata:
+        try:
+            return _generate_network_config_from_imds_metadata(
+                imds_metadata)
+        except Exception as e:
+            msg = 'Failed generating network config from IMDS metadata: %s' % e
+            report_diagnostic_event(msg)
+            LOG.warning(msg)
+    try:
+        return _generate_network_config_from_fallback_config()
+    except Exception as e:
+        msg = 'Failed generating fallback network config: %s' % e
+        report_diagnostic_event(msg)
+        LOG.warning(msg)
+    return {}
```

Comment:
Since you already have logging/telemetry independently for each of the if/else path, I think we can just consolidate this

```suggestion
    try:
        if imds_metadata != sources.UNSET and imds_metadata:
            return _generate_network_config_from_imds_metadata(imds_metadata)
        else:
            return _generate_network_config_from_fallback_config()
    except Exception as e:
        msg = 'Failed generating network config: %s' % e
        report_diagnostic_event(msg)
        LOG.warning(msg)
        return {}
 
```

Reply:
Removed the excessive verbose logging without causing `test_parse_network_config_uses_fallback_cfg_when_no_network_metadata` and `test_parse_network_config_uses_fallback_cfg_when_no_interface_metadata` UTs to fail. This ensures that when an exception is raised during IMDS network config generation, cloud-init will fall back and generate fallback net cfg.